packages/commonwealth/client/scripts/helpers/ContractHelpers/NamespaceFactory.ts

```diff
@@ -262,6 +263,44 @@ class NamespaceFactory extends ContractBase {
       return String(balance / (10 ^ (decimals ?? 18)));
     }
   }
+  /**
+   * mints namespace tokens to assignee on id with desired balance
+   * @param namespace the namespace name
+   * @param id the id on the namespace to mint(admin = 0)
+   * @param desiredBalance the total desired balance(admin = 1)
+   * @param assigneeAddress the address to assign the token to
+   * @param chainId the current chainId
+   * @param walletAddress The senders wallet address
+   * @returns txReceipt
+   * NOTE: If address already has > balance no tokens will be minted
+   */
+  async mintNamespaceTokens(
+    namespace: string,
+    id: number,
+    desiredBalance: number,
+    assigneeAddress: string,
+    chainId: string,
+    walletAddress: string,
+  ) {
+    if (!this.initialized || !this.walletEnabled) {
+      await this.initialize(true, chainId);
+    }
+    const namespaceAddr = await this.getNamespaceAddress(namespace);
+    const namespaceContract = new this.web3.eth.Contract(
+      NamespaceAbi,
+      namespaceAddr,
+    );
+    const balance = await namespaceContract.methods
+      .balanceOf(assigneeAddress, id)
+      .call();
+    const balanceDiff = desiredBalance - Number(balance);
+    if (balanceDiff > 0) {
```

Comment:
is there any logic for `else` scenario?

Reply:
At them moment no, essentially the method aims to get on-chain balance = to the desired balance. If the on-chain balance is already enough the method would basically pass.